### PR TITLE
Feature/auth filtering zaken

### LINKED ISSUE #227: Auth refactoring: filter list of objects
Parent issue - #224

Filtering list of zaken.
* [x] add a method `filter_es (search: Search) -> Search` to `Policy` class to update an ES search on the policy blueprint. 
* [x] In the ES search iterate over all `PermissionDefinition` objects with blueprints and use their `filter_es` methods
* [x] add atomic objects the user has access to ES search
*  ~~If rule predicates are left update the search with the logic from them~~

### PR DESCRIPTION
**Pull request by annashamray**
fixes #227

### CODE REVIEW

**Comment by sergei-maertens** [comment]
@annashamray the diff is a bit too big because of the other PRs, so I'll wait for those to be merged and this one to be rebased before I review this, okay?

**Comment by annashamray** [comment]
@sergei-maertens sure! :)

**Comment by annashamray** [comment]
@sergei-maertens the branch is rebased

**Review by damm89 — APPROVED** [review]
Can't find any obvious mistakes.

File: backend/src/zac/core/services.py [hunk 1]
```diff
@@ -365,8 +365,10 @@ def _get_zaak(zaak_url):
         zaken = list(results)
 
     # resolve zaaktype reference
+    zaaktypen = {zt.url: zt for zt in get_zaaktypen()}
```
Comment (sergei-maertens) [inline backend/src/zac/core/services.py:368]: can you add a TODO for performance optimization in the future -> only fetch the zaaktypen of the zaken that returned in the search results

**Review by sergei-maertens — APPROVED** [review]
:tada: this looks so promising!

File: backend/src/zac/core/services.py (hunk 1 above)
Comment (annashamray) [reply backend/src/zac/core/services.py:369]: Done

**Pull request merged**